src/core/public/i18n/i18n_service.ts

```diff
@@ -17,19 +17,15 @@
  * under the License.
  */
 
-// eslint-disable-next-line @elastic/kibana-custom/no-default-export
-export default function (chrome, internals) {
-  /**
-   * ui/chrome Translations API
-   *
-   *   Translations
-   *     Returns the translations which have been loaded by the Kibana server instance
-   */
+import { i18n } from '@kbn/i18n';
+import { InjectedMetadataStartContract } from '../injected_metadata';
 
-  /**
-   * @return {Object} - Translations
-   */
-  chrome.getTranslations = function () {
-    return internals.translations || [];
-  };
+interface Deps {
+  injectedMetadata: InjectedMetadataStartContract;
+}
+
+export class I18nService {
+  public start({ injectedMetadata }: Deps) {
+    i18n.init(injectedMetadata.getLegacyMetadata().translations);
```

Comment:
It took me a while to track this down and figure out how everything was taking advantage of the translations without consuming the new platform at all, turns out it's because we are just using new platform constructs to store the translations in `@kbn/i18n` module. Based on this I don't think the `I18nService` is necessary, and I don't really think this does what we want for the new platform.

I'm down to move this to the new platform, but if we are going to do that then we should be instantiating something that will wrap the `translations`, and exposing that on the `I18nService` start contract, not by caching the state in `@kbn/i18n`.

In order to address https://github.com/elastic/kibana/pull/23603/files#r225105739 all we would really need to do is move the call to `i18n.init()` out of an angular `.config()` function, and we can just do that by adding it to `ui/chrome/chrome.js` for now until we want to move `@kbn/i18n` fully to the new platform:

```js
import { uiMetadata } from 'ui/metadata';
import i18n from '@kbn/i18n';

i18n.init(uiMetadata.translations);
```

Reply:
@spalger The main reason for this PR is that Notifications [template](https://github.com/elastic/kibana/blob/master/src/core/public/notifications/toasts/toasts_service.tsx#L38) is rendered earlier than `i18n` is initialized with specified locale. 
It's important because to make labels, passing in Notifications, translatable we should wrap Notifications template by `I18nProvider` https://github.com/elastic/kibana/pull/22757/files#diff-53c9b70520405921f846a26c104aeb23R40 which is used to setup the i18n context for a tree. And in the case when we setup the locale in `src/ui/public/i18n/index.js`,  the i18n context for Notifications will be defined with a default locale not counting the data specified in `src/ui/public/i18n/index.js`.

Here the order of files loading:

![image](https://user-images.githubusercontent.com/11524637/47148473-7a83e300-d2d9-11e8-9f9b-b568e063449c.png)

That's why I placed `i18n` initialization just before Notifications template is rendered.

Reply:
Okay, I didn't see that we were going to start consuming `<I18nProvider />` from the new platform. In that case, can we just move the init() call to the entry file so it's initialized before any other code is run? I just don't want to have an i18n service in the new platform unless it's useful, and I don't think this version is really there.

```diff
diff --git a/src/ui/ui_bundles/app_entry_template.js b/src/ui/ui_bundles/app_entry_template.js
index 764e70eaa1..a74ef61848 100644
--- a/src/ui/ui_bundles/app_entry_template.js
+++ b/src/ui/ui_bundles/app_entry_template.js
@@ -33,10 +33,15 @@ import 'whatwg-fetch';
 import 'abortcontroller-polyfill';
 import 'childnode-remove-polyfill';
 
+import { i18n } from '@kbn/i18n';
 import { CoreSystem } from '__kibanaCore__'
 
+const injectedMetadata = JSON.parse(document.querySelector('kbn-injected-metadata').getAttribute('data'));
+i18n.init(injectedMetadata.legacyMetadata.translations);
+
 new CoreSystem({
-  injectedMetadata: JSON.parse(document.querySelector('kbn-injected-metadata').getAttribute('data')),
+  injectedMetadata: injectedMetadata,
   rootDomElement: document.body,
   requireLegacyFiles: () => {
     ${bundle.getRequires().join('\n  ')}
```